src/stream/stream/partial_cmp.rs

```diff
@@ -0,0 +1,56 @@
+use std::cmp::Ordering;
+use std::pin::Pin;
+
+use crate::future::Future;
+use crate::stream::Stream;
+use crate::task::{Context, Poll};
+
+// Lexicographically compares the elements of this `Stream` with those
+// of another.
+#[derive(Debug)]
+pub struct PartialCmpFuture<L: Stream, R> {
+    l: L,
+    r: R,
+}
+
+impl<L: Stream + Unpin, R: Unpin> Unpin for PartialCmpFuture<L, R> {}
+
+impl<L: Stream, R> PartialCmpFuture<L, R> {
+    pub(crate) fn new(l: L, r: R) -> Self {
+        PartialCmpFuture {
+            l,
+            r,
+        }
+    }
+
+    pin_utils::unsafe_pinned!(l: L);
+    pin_utils::unsafe_pinned!(r: R);
+}
+
+impl<L: Stream, R: Stream> Future for PartialCmpFuture<L, R> 
+where 
+    L: Stream + Unpin + Sized,
+    R: Stream + Unpin + Sized,
+    L::Item: PartialOrd<R::Item>
+{
+    type Output = Option<Ordering>;
+
+    fn poll(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
```

Comment:
uhhh, this doesn't look right. Values might arrive from streams at different points at time, but if this happens this doesn't mean that the streams are not equal. The more correct behaviour here would probably be as follows:
1) We cache both elements we are currently comparing in the state of the `PartialCmpFuture`, in `Option`s.
2) As soon as both elements are received (Both options are `.is_some()`) we can compare elements.
3) We have to consider all edge cases: i.e.
 a) we have `l_next`, `l` has closed, but we don't have `r_next` and `r` is not yet closed
 b) the same for `r_next`
 c) both streams are closed but we have only one element left in cache
 d) and so on

Reply:
I see. I wanted to start by replicating the std::iter [code](https://doc.rust-lang.org/1.3.0/src/core/iter.rs.html#3229-3243) for partial_cmp as much as possible, but you're correct that it doesn't account for values from each stream arriving at different times. Thanks for the feedback.

Reply:
I also think that putting `fuse`ing both streams would help.